xbmc/input/KeyboardStat.cpp

```diff
@@ -177,6 +178,15 @@ const CKey CKeyboardStat::ProcessKeyDown(XBMC_keysym& keysym)
   return key;
 }
 
+void CKeyboardStat::ProcessKeyDown(XBMC_keysym& keysym)
+{
+  if (!(m_lastKeysym == keysym))
+  {
+    m_lastKeysym = keysym;
+    m_lastKeyTime = CTimeUtils::GetFrameTime();
```

Comment:
as you are in it you could change GetFrameTime to SystemClockMillis. tying hold time to gui fps makes no sense.